request.go

```diff
@@ -564,10 +564,13 @@ func (r *Request) TraceInfo() TraceInfo {
 	if ct == nil {
 		return TraceInfo{}
 	}
+	if ct.dnsStart.IsZero() {
```

Comment:
Why move this out of `clientTrace.createContext`?  It feels out of place here.  Can you put this in the `ConnectStart` hook where you have the `dnsDone` check?

Reply:
OK, I change it!